lib/mqtt/mqtt.cc

```diff
@@ -235,6 +226,58 @@ namespace
 		return -ETIMEDOUT;
 	}
 
+	/**
+	 * Note from the MQTT 3.1.1 spec:
+	 * The Server MUST allow ClientIds which are between 1 and 23 UTF-8 encoded
+	 * bytes in length, and that contain only the characters
+	 * "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	 *
+	 * Note from us: UTF-8 encoding of 0-9, a-z, A-Z, is 1 Byte per
+	 * character.
+	 */
+	constexpr const char clientIDCharacters[] =
+	  "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	constexpr const int maxClientIDLength = 23;
+
+	/**
+	 * Helper to check if a client ID is valid according to the MQTT
+	 * specification.
+	 */
+	bool is_valid_client_id(const char *id, size_t length)
```

Comment:
This is internal and so should take a `std::string_view`.

Then the code becomes:

```c++
if (id.size() > MQTTMaximumClientIDSize)
{
	return false;
}
for (c in id)
{
	switch (c)
	{
		default:
			return false;
		case '0' ... '9':
		case 'a' ... 'z':
		case 'A' ... 'Z':
			continue;
	}
}
```